rfc/1216-resolve-cloud-tokens.md

```diff
@@ -0,0 +1,845 @@
+the same `QueueName` CloudFormation attribute. This assumption needs verification. If it proves incorrect, 
+we would have to understand which and how many resources behave differently, and either:
+    - Snowflake those resources.
+    - Not support them (i.e let the mechanism fail)
+    - Abandon this solution and go with the [Outputs](#outputs) alternative.
+
+2. Cloud Control doesn't yet support all CloudFormation resources. For example 
+the resource `AWS::SNS::Topic` is not supported (according to this [list][11]). However, 
+the PoC referenced in this RFC does utilize this resource, and everything seems to work fine. 
+Its not clear yet what is going on. Most likely, the list is out-of-date. 
+We need to get an accurate list of unsupported resources and see how this implementation 
+behaves on them. If we see there are too many of them, we might decide to abandon this 
+solution and go with the [Outputs](#outputs) alternative.
+
+3. The AWS CDK token resolver needs to know which stack contains the resource that is
+referenced by the token. Currently, the resolver requires the user explicitly pass the stack
+in its constructor. This is somewhat error prone because the user might mistakenly use a token
+from a different stack. Consider this:
+
+    ```ts
+    const stack1 = new Stack1(...);
+    const stack2 = new Stack2(...);
+
+    // create the resolver with stack1
+    const resolver = new AwsCdkTokenResolver(stack1);
+
+    new kplus.CronJob(manifest, 'CronJob', {
+    containers: [{ 
+      image: 'job',
+      envVariables: {
+        // but pass a token for a resource in stack2
+        BUCKET_NAME: kplus.EnvValue.fromValue(stack2.bucket.bucketName),
+      }
+    }]
+    });
+    ```
+
+    In this case, the implementation will either error out (best case), or 
+    return a value from a different stack in case the in-stack (worst case)
+    logical ids are the same. Ideally we would want to instantiate the resolver
+    with the app instance, not the stack. The resolver would then somehow 
+    automatically identify which stack it needs based on the token. 
+    This might be possible but I haven't yet dived deeper into it. One 
+    thing worth mentioning though is that the AWS CDK logical IDs [**do not contain the 
+    id of the stack**](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/core/lib/stack.ts#L1273), 
+    contradictory to what the documentation](https://docs.aws.amazon.com/cdk/v2/guide/identifiers.html#identifiers_logical_ids) states:
+
+    > *For example, the Amazon S3 bucket in the previous example that is created within 
+    > `Stack2` results in an `AWS::S3::Bucket` resource. The resource's logical ID 
+    > is `Stack2MyBucket4DD88B4F` in the resulting AWS CloudFormation template.*
+
+    This makes it impossible to automatically detect which stack a resource belongs to 
+    based solely on its logical ID.
+
+## Appendix
+
+### `Ref` vs `PhysicalResourceID`
```

Comment:
All the concerns around CC-API and the identifiers make me not like this option.

Reply:
Dropped :)